Library/Homebrew/cmd/desc.rb

```diff
@@ -0,0 +1,59 @@
+require 'descriptions'
+require 'cmd/search'
+
+USAGE =<<EOM
+Print formula descriptions.
+
+Usage:
+
+    brew desc --decache             Delete the cache, if present
+    brew desc --recache             [Re]generate the cache
+    brew desc name [name ...]       Describe the named formula(e)
+    brew desc (-s|-n|-d) <string>   Search specifed fields for string
+    brew desc (-s|-n|-d) /<regex>/  Search specifed fields for regex
+EOM
+
+
+module Homebrew
+  def desc
+    options = ARGV.options_only
+
+    if options.include?("--decache")
+      Descriptions.delete_cache
+      options.delete("--decache")
+      cache_op = true
+    end
+
+    if options.include?("--recache")
+      Descriptions.generate_cache
+      options.delete("--recache")
+      cache_op = true
+    end
+
+    if options.empty?
+      if ARGV.named.empty?
+        print USAGE unless cache_op
```

Comment:
The line works because in Ruby, any assignment to a variable inside of a conditional defines that variable for the remainder of the containing scope — even if the condition is not met. Even if the condition *cannot* be met! If the condition isn’t met, the variable is simply assigned `nil`. For instance the following, if fed into `irb`, will output `nil` and then an error, because after the `if`, `alice` is `nil` and `bob` is undefined.

```ruby
if false
  alice = "Yikes!"
end

puts alice.inspect
puts bob.inspect
```

All of that having been said, I will explicitly set `cache_op` to `false` at the beginning of the method, because relying on that sort of implicit behavior probably amounts to taunting the Elder Gods. 😁